libs/src/sdk/types/File.ts

```diff
@@ -1,8 +1,29 @@
-// Type representing a file in Node environment
-export type NodeFile = {
-  buffer: Buffer
-  name: string
-}
+import { z } from 'zod'
+
+/**
+ * Type representing a file in Node environment
+ */
+const NodeFileSchema = z.object({
+  buffer: z.custom<Buffer>((data: unknown) => data),
+  name: z.optional(z.string())
+})
+export type NodeFile = z.infer<typeof NodeFileSchema>
+
+const BrowserFileSchema = z.custom<File>((data: unknown) => data)
+export type BrowserFile = z.infer<typeof BrowserFileSchema>
 
-// Type representing a file in Node and browser environments
-export type CrossPlatformFile = NodeFile | File
+/**
+ * Type representing a file in Node and browser environments
+ */
+export const CrossPlatformFileSchema = z.union([
```

Comment:
it is so interesting to see how zod works like this! great job promoting it and using it so cleanly